File: fstp-core/resource/com/purefun/fstp/core/bo/otw/TestBO_OTW.java
package com.purefun.fstp.core.bo.otw;

import com.purefun.fstp.core.bo.TestBO;
import com.purefun.fstp.core.bo.pro.TestBO_PRO;
import com.google.protobuf.InvalidProtocolBufferException;
import com.purefun.fstp.core.bo.commom.ICommom_OTW;

public class TestBO_OTW implements ICommom_OTW {
    TestBO_PRO.TestBO.Builder builder = null;
    TestBO bo = null;

    public TestBO_OTW() {
        builder = TestBO_PRO.TestBO.newBuilder();
        bo= new TestBO();
        builder.setUuid(bo.uuid);
        builder.setBoid(bo.boid);
        builder.setDestination(bo.destination);
    }

    public TestBO_OTW(byte[] message) throws InvalidProtocolBufferException {
        builder = TestBO_PRO.TestBO.newBuilder();
        bo= new TestBO();
        TestBO_PRO.TestBO receive = TestBO_PRO.TestBO.parseFrom(message);
        setUuid(receive.getUuid());
        setBoid(receive.getBoid());
        setDestination(receive.getDestination());
        setServername(receive.getServername());
        setMsg(receive.getMsg());
    }

    @Override
    public TestBO_PRO.TestBO.Builder getBuilder() { 
        return builder;
    }

    @Override
    public TestBO getBo() { 
        return bo;
    }

    public java.lang.String getUuid() {
        return builder.getUuid();
    }

    public void setUuid(java.lang.String uuid) {
        bo.uuid = uuid;
        builder.setUuid(uuid);
    }

    public long getBoid() {
        return builder.getBoid();
    }

    public void setBoid(long boid) {
        bo.boid = boid;
        builder.setBoid(boid);
    }

    public java.lang.String getDestination() {
        return builder.getDestination();
    }

    public void setDestination(java.lang.String destination) {
        bo.destination = destination;
        builder.setDestination(destination);
    }

    public java.lang.String getServername() {
        return builder.getServername();
    }

    public void setServername(java.lang.String servername) {
        bo.servername = servername;
        builder.setServername(servername);
    }

    public java.lang.String getMsg() {
        return builder.getMsg();
    }

    public void setMsg(java.lang.String msg) {
        bo.msg = msg;
        builder.setMsg(msg);
    }

    public String toString() {
        return "TestBO_OTW ["+
            "uuid = " + getUuid() +"," +
            "boid = " + getBoid() +"," +
            "destination = " + getDestination() +"," +
            "servername = " + getServername() +"," +
            "msg = " + getMsg() +"," +
    "]";
    }
}

File: fstp-python/core/bo/otw/TestBO_OTW.py
from core.bo.pro import TestBO_pb2
from core.bo.model.TestBO import TestBO


class TestBO_OTW(object):

    def __init__(self, byteMsg = None):
        self._bo_pro = TestBO_pb2.TestBO()
        self._bo = TestBO()      
        
        if byteMsg is not None:
            self._bo_pro.ParseFromString(byteMsg)
            self.__setDataFromPB()
        else:
            self.__setDataFromBO()
                  
    def __setDataFromBO(self):  
        self._bo_pro.boid = self._bo.boid
        self._bo_pro.uuid = self._bo.uuid
        self._bo_pro.destination = self._bo.destination 
        
    def __setDataFromPB(self):  
        self._bo.boid = self._bo_pro.boid
        self._bo.uuid = self._bo_pro.uuid
        self._bo.destination = self._bo_pro.destination 
        self._bo.servername = self._bo_pro.servername
        self._bo.msg = self._bo_pro.msg
  
    
    def getBO(self):
        return self._bo
    
    def getProBO(self):
        return self._bo_pro
    
    def setDestination(self, destination):
        self._bo.destination = destination
        self._bo_pro.destination = destination
    
    def setMsg(self, msg):
        self._bo.msg = msg
        self._bo_pro.msg = msg

          
test = TestBO_OTW() 
test.setMsg('123456789') 
# print test.getBO().uuid 
# print test.getBO().destination
# print test.getBO().msg
# print
# 
# print test.getProBO().uuid 
# print test.getProBO().destination
# print test.getProBO().msg
# print
# 
rec = TestBO_OTW(test.getProBO().SerializeToString())
 
print rec.getBO().uuid 
print rec.getBO().destination
print rec.getBO().msg
print
# 
# print rec.getProBO().uuid 
# print rec.getProBO().destination
# print rec.getProBO().msg
# print        
        